tests/nopytest_test_coadds.py

```diff
@@ -80,72 +80,24 @@
 DATAREPO_ROOT = os.path.join(os.path.dirname(__file__), ".tests", "testCoadds-data")
 
 
+def assertWrapper(func):
+    """Decorator to intercept AssertionError and reraise whilst recording
+    a failure."""
```

Comment:
I think it'd be worth writing just another sentence here describing why this is a useful thing to do, just to help the reader orient themselves. Something along the lines of “this enables us to track whether any tests failed, and if so to preserve the test data for future reference”.